pkg/certgenerator/v1beta1/generator.go

```diff
@@ -64,26 +68,54 @@ var _ manager.Runnable = &CertGenerator{}
 var _ manager.LeaderElectionRunnable = &CertGenerator{}
 
 func (c *CertGenerator) Start(ctx context.Context) error {
-	if err := c.generate(ctx); err != nil {
+	if c.enableGenerateCerts {
+		if err := c.generate(ctx); err != nil {
+			return err
+		}
+	}
+	// We need to wait for certs to get ready even though using cert-manager.
+	klog.Info("Waiting for certs to get ready.")
+	if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
```

Comment:
@johnugeorge @tenzen-y Does it look confusing that we have this timeout in our generator package even if Katib Cert Generator doesn't used ?
Maybe we can add this timeout to the [`setupControllers` section before we adding the Webhook to the manager](https://github.com/kubeflow/katib/blob/69e9dfea1dea28dc1f5807226f7425efb31b71b6/cmd/katib-controller/v1beta1/main.go#L167-L170) ?

We can rename `certsReady` channel to `certsGenerated` to be more precise.

Reply:
I think locating this timeout here would be better since this timeout is a function for the webhook certs.
However, I agree with your confusion. So renaming the `certgenerator` with `certmanager` or `certcontroller` might be better to clarify the purpose of this component.

@andreyvelich WDYT?

Reply:
I think `certcontroller` would be better since `certmanager` conflicts context with [`cert-manager`](https://cert-manager.io/) :-/

Reply:
If we rename it to `certcontroller` how we can deal with Katib Config Settings ?
If user wants the enable cert generator they set the following setting:
```
  certGenerator:
    enable: true
```

Reply:
So, the `certcontroller` will start only in the standalone installation.

Reply:
I think, if we are going to remove this timeout from Kubeflow installation, we need to name it as `certgenerator` since it is only internal logic that we have when Katib `certGenerator` is enabled.
Let's keep it as `certgenerator` for now and use this timeout only when our internal Cert Generator is used.
Does it sound good @tenzen-y ?

Reply:
@andreyvelich That sounds good to me. Thanks for the great suggestion!

Reply:
Done.